hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieCompactor.java

```diff
@@ -24,20 +25,43 @@
 import com.uber.hoodie.table.HoodieTable;
 import org.apache.spark.api.java.JavaSparkContext;
 
+import java.io.IOException;
 import java.io.Serializable;
 import java.util.Date;
+import java.util.List;
 
 /**
  * A HoodieCompactor runs compaction on a hoodie table
  */
 public interface HoodieCompactor extends Serializable {
 
   /**
-   * Compact the delta files with the data files
+   * Compact the delta files with the data files with autoCommit
```

Comment:
This is used again in inline compaction. We can get rid of this and use the 2nd API and move the writing to metadata logic in the writeClient, this is an option.